### PR TITLE
Check for (and fix) whitespace formatting in non-C++ files

### PR DESCRIPTION
**Pull request by devbww**
Add a phase in ci/check-style.sh that fixes whitespace formatting
in files not handled by clang-format.  For now this simply removes
trailing blanks.

Also build paths to ignore in style checks by reading .gitignore.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2522)

### CODE REVIEW

File: ci/check-style.sh [hunk 1]
```diff
@@ -25,36 +25,39 @@ fi
...
 
-find google/cloud -name '*.h' -print0 \
-  | xargs -0 awk -f "${BINDIR}/check-include-guards.gawk"
+# Build paths to ignore in find(1) commands by reading .gitignore.
+declare -a ignore=( -path ./.git )
+if [ -f .gitignore ]; then
+  while read -r line; do
+    case "${line}" in
+    [^#]*/*) ignore+=( -o -path "./$(expr "${line}" : '\(.*\)/')" );;
+    [^#]*)   ignore+=( -o -name "${line}" );;
```
Comment (remyabel) [inline ci/check-style.sh]: Should this be `"./${line}"`?

File: ci/travis/build-docker.sh [hunk 2]
```diff
@@ -65,7 +65,7 @@ fi
```
Comment (remyabel) [inline ci/travis/build-docker.sh]: Accidental permission change?

**Comment by remyabel** [comment]
I am not 100% sure, but I think that for `find . \( "${ignore[@]}" \)`, the array will expand to something like ` -name bazel-* -o -name *~`. Notice the lack of quotes. We probably want to quote them to prevent shell globbing.

File: ci/check-style.sh (hunk 1 above)
Comment (devbww) [reply ci/check-style.sh]: A slash in a `-name` argument will cause it to never match.  That is, `find -name` is only looking at the basename of the candidate path.

In the case of `bazel-*` from our `.gitignore`, we should probably be saying `bazel-*/'.

All that said, `find` can't implement all the `.gitignore` functionality (`**` is a simple example), so this mapping is certainly based on us only using simple "ignore" specifications.

**Comment by remyabel** [comment]
Actually, disregard my last comment. It seems that `.gitignore` uses `*` for globbing, not regex, so leaving the expressions unquoted will be consistent.

File: ci/travis/build-docker.sh (hunk 2 above)
Comment (devbww) [reply ci/travis/build-docker.sh]: Thanks.  Fixed.

**Comment by devbww** [comment]
> I am not 100% sure, but I think that for find . \( "${ignore[@]}" \), the array will expand to something like -name bazel-* -o -name *~. Notice the lack of quotes. We probably want to quote them to prevent shell globbing.

I believe it will expand to the equivalent of (`"-name" "bazel-*" "-o" "-name" "*~"`), so there is no risk of those arguments being re-expanded.  That is, they are quoted.  For example, ...

```
$ echo C*
CMakeLists.txt CODE_OF_CONDUCT.md CONTRIBUTING.md
$ declare -a A=( 'C*' )
$ echo "${A[@]}"
C*
$ echo ${A[@]}
CMakeLists.txt CODE_OF_CONDUCT.md CONTRIBUTING.md
```

**Comment by coryan** [comment]
FWIW, the UBSAN build seems to have finished, but Travis may have lost the notification.  Feel free to merge if you want.

**Comment by devbww** [comment]
I'll wait to see if @remyabel has anything to add.

**Comment by remyabel** [comment]
Nope it LGTM. Sorry for the mistaken comments.

**Pull request merged**